src/components/appointments/shop/appointmentCard.tsx

```diff
@@ -0,0 +1,143 @@
+import AppointmentTypes from "@redux/types/appointmentTypes";
+import styles from "@styles/pages/appointments/ShopAppointments.module.css";
+import classNames from "classnames";
+import { Button } from "primereact/button";
+import { Card } from "primereact/card";
+import { useDispatch } from "react-redux";
+import { AppointmentStatus, IAppointment } from "../../../types/appointment";
+
+interface IAppointmentCardProps {
+  appointment: IAppointment;
+  appointmentProgress: AppointmentStatus;
+}
+
+const AppointmentCard = (props: IAppointmentCardProps) => {
+  const dispatch = useDispatch();
+
+  const { appointment, appointmentProgress } = props;
+
+  const handleButtonClick = (
+    appointment: IAppointment,
+    status: AppointmentStatus
+  ): void => {
+    const payload = { id: appointment.id, status: status };
+    dispatch({ type: AppointmentTypes.SET_APPOINTMENT_STATUS, payload });
```

Comment:
```suggestion
     dispatch(setAppointmentStatus({ id: appointment.id, status: status }));
```

Dispatch the action like this.